cub/block/block_adjacent_difference.cuh

```diff
@@ -39,27 +41,105 @@
 
 CUB_NAMESPACE_BEGIN
 
-template <
-    typename    T,
-    int         BLOCK_DIM_X,
-    int         BLOCK_DIM_Y     = 1,
-    int         BLOCK_DIM_Z     = 1,
-    int         PTX_ARCH        = CUB_PTX_ARCH>
+
+/**
+ * @brief BlockAdjacentDifference provides
+ *        [<em>collective</em>](index.html#sec0) methods for computing the
+ *        differences of adjacent elements partitioned across a CUDA thread
+ *        block.
+ *
+ * @ingroup BlockModule
+ *
+ * @par Overview
+ * - BlockAdjacentDifference calculates the differences of adjacent elements in
+ *   the elements partitioned across a CUDA thread block. Because the binary
+ *   operation could be noncommutative, there are two sets of methods.
+ *   Methods named SubtractLeft subtract left element `i - 1` of input sequence
+ *   from current element `i`. Methods named SubtractRight subtract current
+ *   element `i` from the right one `i + 1`:
+ *   @par
+ *   @code
+ *   int values[4]; // [1, 2, 3, 4]
+ *   //...
+ *   int subtract_left_result[4];  <-- [  1,  1,  1,  1 ]
+ *   int subtract_right_result[4]; <-- [ -1, -1, -1,  4 ]
+ *   @endcode
+ * - For SubtractLeft, if the left element is out of bounds, the
+ *   output value is assigned to `input[0]` without modification.
+ * - For SubtractRight, if the right element is out of bounds, the output value
+ *   is assigned to the current input value without modification.
+ * - The following example under the examples/block folder illustrates usage of
+ *   dynamically shared memory with BlockReduce and how to re-purpose
+ *   the same memory region:
+ *   <a href="../../examples/block/example_block_reduce_dyn_smem.cu">example_block_reduce_dyn_smem.cu</a>
+ *   This example can be easily adapted to the storage required by
+ *   BlockAdjacentDifference.
+ *
+ * @par Snippet
+ * The code snippet below illustrates how to use @p BlockAdjacentDifference to
+ * compute the left difference between adjacent elements.
+ *
+ * @par
+ * @code
+ * #include <cub/cub.cuh>
+ * // or equivalently <cub/block/block_adjacent_difference.cuh>
+ *
+ * struct CustomDifference
+ * {
+ *   template <typename DataType>
+ *   __device__ DataType operator()(DataType &lhs, DataType &rhs)
+ *   {
+ *     return lhs - rhs;
+ *   }
+ * };
+ *
+ * __global__ void ExampleKernel(...)
+ * {
+ *     // Specialize BlockAdjacentDifference for a 1D block of
+ *     // 128 threads on type int
```

Comment:
s/on/of/

(this is repeated in a few other docstrings)

Reply:
There are quite a few places with this typo:
```
rg "threads on type" | wc -l
33
```

I'll fix this in the `block_scan`, `block_discontinuity` and `block_reduce` documentation as well.